### PR TITLE
Closes #1728 Adds footnote to scatterplot widget

### LINKED ISSUE #1728: Feature: Add footnote to scatter plot widget that describes the sizing of the points.
## Feature Details
Footnote appears at the bottom of the scatter plot.

## Additional Comments
If group metadata is missing footnote shouldn't appear.

### PR DESCRIPTION
**Pull request by zdz2101**
## Overview
Adds code for footnote at the bottom of scatterplot to indicate size of points is relative to size of group

## Connected Issues

- Closes #1728

### CODE REVIEW

File: R/Widget_ScatterPlot.R [hunk 1]
```diff
@@ -48,6 +48,7 @@ Widget_ScatterPlot <- function(
     dfBounds = dfBounds,
     bAddGroupSelect = bAddGroupSelect,
     strShinyGroupSelectID = strShinyGroupSelectID,
+    strFootnote = ifelse(bAddGroupSelect, "Size of the point is relative to size of the group.", ""),
```
Comment (zdz2101) [inline R/Widget_ScatterPlot.R:51]: @samussiah in no universe am I a JS writer, but this seemed simple enough to take as an initial task, is the logical `bAddGroupSelect` the if condition you meant in #1728 ?

File: inst/htmlwidgets/Widget_ScatterPlot.js [hunk 2]
```diff
@@ -30,6 +30,17 @@ HTMLWidgets.widget({
                     input.dfGroups,
                     input.bAddGroupSelect
                 );
+
+                // Add a footnote below the scatter plot.
+                const footnote = document.createElement('div'); // Create a div for the footnote.
+                footnote.style.fontSize = '10px'; // Set a smaller font size for the footnote.
```
Comment (zdz2101) [inline inst/htmlwidgets/Widget_ScatterPlot.js:36]: and not sure how to do this dynamically but I think this gets us on the board/working version?

Ran example 3.3 to render below
image

File: R/Widget_ScatterPlot.R (hunk 1 above)
Comment (samussiah) [reply R/Widget_ScatterPlot.R:51]: You're on the right track but the logic should be around `dfGroups` instead of `bAddGroupSelect`.

Comment (samussiah) [inline R/Widget_ScatterPlot.R:51]:
```suggestion
    strFootnote = ifelse(!is.null(dfGroups), "Point size is relative to the number of enrolled participants.", ""),
```

File: inst/htmlwidgets/Widget_ScatterPlot.js (hunk 2 above)
Comment (samussiah) [reply inst/htmlwidgets/Widget_ScatterPlot.js:36]: Love it, nice and subtle.

**Review by samussiah — CHANGES_REQUESTED** [review]
Minor update but this is exactly what I had in mind.

**Review by samussiah — APPROVED** [review]
💯

**Pull request merged**